_tile_downloader.py

```diff
@@ -0,0 +1,231 @@
+import tempfile
+import time
+from pathlib import Path
+from typing import Union, Tuple, Type, Optional
+
+import cv2
+import numpy as np
+import rasterio as rio
+import rasterio.mask
+import rasterio.warp
+import requests
+from darkgeotile import BaseTile
+from pyproj import transform, Proj
+from shapely.geometry import Polygon
+
+import maps
+from utils import ImageFormat, get_expected_path
+
+
+def download_tiles(
+        map_: Type[maps.Map],
```

Comment:
Try to not use underscores in function args. For ex: you can rename it to tile_source or smth like that